GRG/Widget.js

```diff
@@ -1731,30 +1742,37 @@ define([
                     polysToLoop = [];
                     polysToLoop = polys;
                     currentValue = currentValue / 10;                
-                  }
-                  var count = 1;
+                  }                  
                   for (i = 0; i < polysToLoop.length; i++) {
                     if (this.map.spatialReference.wkid !== 4326) {
                       var graphic = new Graphic(polysToLoop[i].clippedPolyToUTMZone);
                     } else {
                       var graphic = new Graphic(WebMercatorUtils.webMercatorToGeographic(polysToLoop[i].clippedPolyToUTMZone));
                     }
-                    var label = this.grgPointByRefLabelFormat.value
-                    label = label.replace(/Y/, polysToLoop[i].y);
-                    label = label.replace(/X/, polysToLoop[i].x); 
-                    label = label.replace(/S/, polysToLoop[i].GZD); 
-                    label = label.replace(/Z/, polysToLoop[i].utmZone + polysToLoop[i].latitudeZone);                   
-                    graphic.setAttributes({'grid': label});                  
-                    features.push(graphic);
+                    if(this._labelTypeWithRefSys === 'gridReferenceSystem') {
```

Comment:
'gridReferenceSystem' can either be a pseudo const variable at the class scope or part of an object scoped at class level as well.